test/Exception.js

```diff
@@ -174,14 +161,46 @@ describe('Exception', () => {
 		})
 	})
 
+	it('should recover from error boundary in the event of an exception', () => {
+		const target = document.createElement('div')
+		const Primary = props => {
+			const [error, dispatch] = useState('')
+
+			return h(Boundary, {fallback: error => dispatch(error.message)}, error ? error : h(Secondary))
+		}
+		const Secondary = props => {
+			throw 1
+		}
+
+		render(h('div', {}, h(Primary)), target, (current) => {
+			assert.html(current, '<div>1</div>')
+		})
+	})
+
+	it('should throw within promise', (done) => {
+		const target = document.createElement('div')
+		const stack = assert.spyr(console, 'error')
+		const Primary = props => {
+			return stack.current = Promise.reject(1)
+		}
+
+		render(h(Primary), target, () => {}).then((current) => {
+			stack.push('error')
```

Comment:
this should be `done(new Error ...)`

Reply:
See the next assertion, this is not meant to run, which is why we don't have an `"error"` value in the `deepEqual` assertion array. Maybe it should be just `assert.fail()`.

Reply:
yes, this was just a nit so the test fails instead of timing out